pkg/arvo/lib/pull-hook.hoon

```diff
@@ -286,6 +311,58 @@
     --
   |_  =bowl:gall
   +*  og   ~(. pull-hook bowl)
+  ++  mule-scry
+    |=  [ref=* raw=*]
+    =/  pax=(unit path)
+      ((soft path) raw)
+    ?~  pax  ~
+    ?.  ?=([@ @ @ @ *] u.pax)  ~
+    =/  ship
+      (slaw %p i.t.u.pax)
+    =/  ved
+      (slay i.t.t.t.u.pax)
+    =/  dat
+      ?~  ved  now.bowl
+      =/  cas=(unit case)
+        ((soft case) p.u.ved)
+      ?~  cas  now.bowl
+      ?:  ?=(%da -.u.cas)
+        p.u.cas
+      now.bowl
+    ::  catch bad gall scries early
+    ?:  ?&  =((end 3 1 i.u.pax) %g)
+            ?|  !=(`our.bowl ship)
+                !=(dat now.bowl)
+            ==
+        ==
+      ~
+    ``.^(* u.pax)
+  ++  handle-kick
+    |=  [rid=resource =ship]
+    ^-  (quip card _state)
+    =/  res=toon
+      (mock [|.((on-pull-kick:og rid)) %9 2 %0 1] mule-scry)
+    =/  pax=(unit path)
+      !<  (unit path) 
+      :-  -:!>(*(unit path)) 
+      ?:(?=(%0 -.res) p.res ~)
```

Comment:
That's some spicy hoon.  Seems legit, though, as in Gall.  It's the only way to intercept a scry and get a typed result afaik.

Reply:
Shouldn't that be `[~ p.res]`, assuming the product is a path?

Reply:
Also, if you're only every scrying for a path, it should be fine to skip the evil-vase construction and just clam.

Reply:
It produces a unit path, he has no need to distinguish between the app producing null or crashing.

Reply:
Ok. I'd still encourage clamming.